### PR TITLE
Hotspot Metadata

### PR DESCRIPTION
**Pull request by ThornM9**
#22

### CODE REVIEW

File: packages/helium-entity-manager-sdk/src/resolvers.ts [hunk 1]
```diff
@@ -2,25 +2,15 @@ import { PublicKey } from "@solana/web3.js";
...
-        return new PublicKey("credacwrBVewZAgCwNgowCSMbCiepuesprUWPBeLTSg");
-      case "tokenMetadataProgram":
-        return new PublicKey("metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s");
-      case "heliumSubDaosProgram":
-        return new PublicKey("hdaojPkgSD8bciDc1w2Z4kXFFibCXngJiw2GRpEL7Wf");
-      default:
-        return;
-    }
-  }),
+  heliumCommonResolver,
```
Comment (bryzettler) [inline packages/helium-entity-manager-sdk/src/resolvers.ts:13]: Nice

File: programs/helium-entity-manager/src/state.rs [hunk 2]
```diff
@@ -10,6 +10,10 @@ pub struct HotspotConfigV0 {
   pub authority: Pubkey,
   pub symbol: String,
   pub sub_dao: Pubkey,
+  pub min_gain: i32,
+  pub max_gain: i32,
+  pub full_location_staking_fee: u64,
```
Comment (bryzettler) [inline programs/helium-entity-manager/src/state.rs:15]: `staking_fee` is throwing me off. Should it just be `dc_fee`? are they actually staking something?

**Review by bryzettler — APPROVED** [review]

File: programs/helium-entity-manager/src/state.rs (hunk 2 above)
Comment (ThornM9) [reply programs/helium-entity-manager/src/state.rs:15]: It's a misnomer but that's the term used by Helium

File: programs/helium-entity-manager/src/instructions/metadata/change_metadata_v0.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  pub hotspot_owner: Signer<'info>,
+  #[account(
+    associated_token::mint = hotspot,
+    associated_token::authority = hotspot_owner,
+    constraint = owner_hotspot_ata.amount == 1,
+  )]
+  pub owner_hotspot_ata: Box<Account<'info, TokenAccount>>,
+}
+
+pub fn handler(ctx: Context<ChangeMetadataV0>, args: ChangeMetadataArgsV0) -> Result<()> {
```
Comment (ChewingGlass) [inline programs/helium-entity-manager/src/instructions/metadata/change_metadata_v0.rs:34]: We shouldn't allow on chain changes without the burn, right? What's the purpose of the two-stage? Could collapse this all to one ix right?

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

File: programs/helium-entity-manager/src/instructions/metadata/change_metadata_v0.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ThornM9) [reply programs/helium-entity-manager/src/instructions/metadata/change_metadata_v0.rs:34]: Following this proposal: https://gist.github.com/madninja/4dabc2db8bdf698df11fa3ea960bb9d7
Not entirely sure of the reasoning

File: packages/helium-cli/src/bootstrap.ts [hunk 4]
```diff
@@ -188,7 +188,6 @@ async function run() {
     provider,
     mintKeypair: dcKeypair,
     amount: argv.numDc,
-    decimals: 0,
```
[the remark this answers is not in the document]
Comment (ThornM9) [reply packages/helium-cli/src/bootstrap.ts:191]: Done

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**